src/ajax.js

```diff
@@ -454,17 +454,19 @@ jQuery.extend( {
 
 				// Builds headers hashtable if needed
 				getResponseHeader: function( key ) {
-					var match;
+					var match, existingHeaderValues, headerValues;
 					if ( completed ) {
 						if ( !responseHeaders ) {
 							responseHeaders = {};
 							while ( ( match = rheaders.exec( responseHeadersString ) ) ) {
-								responseHeaders[ match[ 1 ].toLowerCase() ] = match[ 2 ];
+								existingHeaderValues = responseHeaders[ match[ 1 ].toLowerCase() ];
+								headerValues = ( existingHeaderValues || [] ).concat( match[ 2 ] );
```

Comment:
Our existing code already [misbehaves](https://jsfiddle.net/0cj8y357/) when someone attempts to retrieve a header that shares the name of a lowercase `Object.prototype` property (e.g., "constructor" or "\_\_proto__"), but this will actually throw. If we're making changes, I'd like to fix that issue rather than exacerbate it (e.g., by appending a space as in [event.js](https://github.com/jquery/jquery/blob/e743cbd28553267f955f71ea7248377915613fd9/src/event.js#L390)).

Reply:
#4214

Reply:
Good catch! I didn't think of Object.prototype when using the same pattern on some of the stuff I've been doing, makes for a good interview question hehe

Fixed.